flytekit/image_spec/image_spec.py

```diff
@@ -181,6 +183,16 @@ def with_apt_packages(self, apt_packages: Union[str, List[str]]) -> "ImageSpec":
 
         return new_image_spec
 
+    def force_push(self) -> "ImageSpec":
+        """
+        Builder that returns a new image spec with force push enabled.
+        """
+        new_image_spec = copy.deepcopy(self)
+        new_image_spec._is_force_push = True
+        self._is_force_push = True
```

Comment:
```suggestion
```

Reply:
updated

Reply:
I found that if `self._is_force_push = True` is removed, even if `force_push()` is used, `is_force_push` will still be `False` (unchanged) on [this conditional](https://github.com/jasonlai1218/flytekit/blob/add-image-spec-force-push/flytekit/image_spec/image_spec.py#L235), and the result is that `skip building`

Reply:
```
def force_push(self) -> "ImageSpec":
        """
        Builder that returns a new image spec with force push enabled.
        """
        new_image_spec = copy.deepcopy(self)
        new_image_spec._is_force_push = True
        print(f"force_push: {vars(new_image_spec)}")

        return new_image_spec
```

```
 @classmethod
    @lru_cache
    def build(cls, image_spec: ImageSpec) -> str:
        print(f"build: {vars(image_spec)}")
        ...
        ...
```


> force_push: {'name': 'test_version_1', 'python_version': None, 'builder': None, 'source_root': None, 'env': {'FLYTE_SDK_LOGGING_LEVEL': '20'}, 'registry': 'jasonlai1218', 'packages': ['git+https://github.com/jasonlai1218/flytekit.git@add-image-spec-force-push#commit=d3a0fb5e20d1dad4d43f34939efafda920c598fb'], 'conda_packages': None, 'conda_channels': None, 'requirements': None, 'apt_packages': ['git'], 'cuda': None, 'cudnn': None, 'base_image': None, 'platform': 'linux/amd64', 'pip_index': None, 'registry_config': None, 'commands': None, '_is_force_push': True}
>
> Running Execution on Remote.
> 
> build: {'name': 'test_version_1', 'python_version': None, 'builder': None, 'source_root': None, 'env': {'FLYTE_SDK_LOGGING_LEVEL': '20'}, 'registry': 'jasonlai1218', 'packages': ['git+https://github.com/jasonlai1218/flytekit.git@add-image-spec-force-push#commit=d3a0fb5e20d1dad4d43f34939efafda920c598fb'], 'conda_packages': None, 'conda_channels': None, 'requirements': None, 'apt_packages': ['git'], 'cuda': None, 'cudnn': None, 'base_image': None, 'platform': 'linux/amd64', 'pip_index': None, 'registry_config': None, 'commands': None, '_is_force_push': False}

Reply:
```
py_image_spec = ImageSpec(
    name="test_version_1",
    registry="jasonlai1218",
)
py_image_spec = py_image_spec.force_push()
```
If used like this, it can work

Reply:
or
```
py_image_spec = ImageSpec(
    name="test_version_1",
    registry="jasonlai1218",
).force_push()
```